timestring/Range.py

```diff
@@ -169,14 +180,41 @@ def __init__(self, start, end=None, offset=None, start_of_week=0, tz=None, verbo
                             start += '1 month'
                     end = start + '1 month'
 
-                elif group.get('year_5'):
-                    # a whole year
+                elif group['date_5'] or group['date_6'] or group['time_2']:
+                    # TODO: Move this code to Date?
+                    year = g('year', 'year_2', 'year_3', 'year_4', 'year_5', 'year_6')
+                    month = g('month', 'month_2', 'month_3', 'month_4', 'month_5')
+                    day = g('date', 'date_2', 'date_3', 'date_4')
+                    hour = g('hour', 'hour_2', 'hour_3')
+                    minute = g('minute', 'minute_2')
+                    second = g('seconds')
+
                     start = Date(start, offset=offset, tz=tz)
-                    start = start.replace(day=1, month=1, hour=0, minute=0, second=0)
-                    end = start + '1 year'
+                    if month is None and year is not None:
+                        start = start.replace(month=1)
+                    if day is None and (month or not (hour or group['daytime'])):
+                        start = start.replace(day=1)
+                    if hour is None and not group['daytime']:
+                        start = start.replace(hour=0)
+                    if minute is None:
+                        start = start.replace(minute=0)
+                    if second is None:
+                        start = start.replace(second=0)
+
+                    if second:
+                        end = start + '1 second'
+                    elif minute:
+                        end = start + '1 minute'
+                    elif hour:
+                        end = start + '1 hour'
+                    elif day:
+                        end = start + '1 day'
```

Comment:
By mistake I deleted @dpathania1 's comment that I was replying to. The comment asked shouldn't years ago be 1 day long. That change is in https://github.com/iAmPlus/timestring/pull/18, but I can put both of them in one PR if that's easier to review.